fuzz/fuzz_targets/refcounted_model.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2021-2022 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or MIT.
+
+//! Model check with reference counting:
+//! Checks that a sequence of operations and restarts behaves the same as an in-memory collection.
+
+#![no_main]
+use arbitrary::Arbitrary;
+use libfuzzer_sys::fuzz_target;
+use std::collections::btree_map::{BTreeMap, Entry};
+use tempfile::tempdir;
+
+#[derive(Arbitrary, Debug, Clone, Copy)]
+enum Operation {
+	Set(u8, u8),
+	Dereference(u8),
+	Reference(u8),
+}
+
+#[derive(Arbitrary, Debug)]
+enum Action {
+	Transaction(Vec<Operation>),
+	Restart,
+}
+
+#[derive(Arbitrary, Debug, Clone, Copy)]
+enum CompressionType {
+	NoCompression,
+	Snappy,
+	Lz4,
+}
+
+#[derive(Arbitrary, Debug)]
+struct Config {
```

Comment:
We could also use different fuzz_target for different config, but if config get complex it won't be manageable so sounds ok.
(if needed it would not take long to use a static config)

Reply:
Yes, might be nice. But I believe we will quickly want to play with various thresholds (queue size...) to do deeper testing and so a dynamic configuration makes more sense. What do you think about it?